apt/minimization/minimizer.py

```diff
@@ -140,7 +147,7 @@ def fit_transform(self, X=None, y=None):
         self.fit(X, y)
         return self.transform(X)
 
-    def fit(self, X=None, y=None):
+    def fit(self, X: Union[np.ndarray, pd.DataFrame] = None, y=None):
```

Comment:
Please add this Union typing to the y param as well, and to the fit_transform method too.